modules/study/src/main/Socket.scala

```diff
@@ -146,10 +170,16 @@ private final class Socket(
       addMember(uid.value, member)
       notifyCrowd
       sender ! Socket.Connected(enumerator, member)
+      userId match {
+        case Some(u) => sendStudyJoin(u)
+        case None => {}
+      }
 
     case Quit(uid) =>
       members get uid foreach { member =>
         quit(uid)
+        val userId = member.userId.get
```

Comment:
Never, ever, use `.get` on an `Option`. It will throw an exception (boo) if `None`. Scala sucks for even having that function.

Instead, `member.userId foreach sendStudyQuit`